packages/cozy-client/src/models/konnectorFolder.js

```diff
@@ -0,0 +1,213 @@
+  /**
+   * Now that we have our sanitizedPath, we can split it in two strings
+   * * `baseDir` containing the baseDir path
+   * * `buildedSubDir` containing the rest of the path (ie the path without baseDir)
+   */
+  const baseDir = sanitizedPath.split('/', 1)
+  const buildedSubDir = buildSubDir(sanitizedPath, baseDir[0])
+
+  const renderedBaseDir = renderBaseDir(baseDir[0], folders)
+  const renderedPath = renderSubDir(buildedSubDir, {
+    // When adding a new allowed variable here, please keep documentation
+    // of `renderSubDir` function up to date.
+    konnector: konnector.name,
+    account: accountModel
+      .getAccountName(account)
+      .replace(sanitizeAccountIdentifierRx, '-')
+  })
+  return `/${renderedBaseDir}/${renderedPath}`
+}
+
+/**
+ * Check if the provided Path start withs our allowedBaseDirPath to see
+ *
+ * @param {String} path - path to test
+ * @returns {Boolean}
+ */
+const hasBaseDir = path => {
+  return allowedBaseDirVariables.some(baseDirVar => {
+    return path.startsWith(baseDirVar)
+  })
+}
+/**
+ * Base directories are directory where konnector may copy their data.
+ * They are expressed as variables which then need to be localized.
+ * Default is `$administrative`.
+ */
+const allowedBaseDirVariables = ['$administrative', '$photos']
+
+/**
+ * This method creates the subDir. We can't have an empty subDir, so we set
+ * it to our default '$konnector/$account'
+ *
+ * @param {String} fullPath String containing potentially the defaultDir
+ * @param {String} defaultDir String to remove from the fullPath
+ */
+const buildSubDir = (fullPath, defaultDir) => {
+  let buildedSubDir = fullPath.substring(defaultDir.length)
+  if (buildedSubDir === '') {
+    buildedSubDir = '$konnector/$account'
+  }
+  return buildedSubDir
+}
+
+/**
+ * Render base directory, based on given folders object.
+ * For example, it will render `$administrative` with the given value passed in
+ * folders object. We expect to find in folders a localized value.
+ *
+ * @param  {String} baseDir base directory variable, expects `$administrative`
+ * or `$photos`
+ * @param  {Object} folders Object indexing base directory variable with
+ * corresponding localized name.
+ * @returns {String}         Localized directory
+ */
+const renderBaseDir = (baseDir, folders = {}) => {
+  // Look for variable name into folders but without $ prefix
+  const renderedBaseDir =
+    folders[baseDir.slice(1)] || DEFAULT_LOCALIZED_BASE_DIR
+  // Trim `/` and avoid multiple `/` characters with regexp
+  return trim(renderedBaseDir.replace(/(\/+)/g, '/'), '/')
+}
+
+// Default name for base directory
+const DEFAULT_LOCALIZED_BASE_DIR = 'Administrative'
+
+/**
+ * Render the given folder path using the given `variables` object.
+ * Available variable are `$konnector` (konnector name) and `$account`
+ * (account label, i.e. id or name)
+ *
+ * @param  {String} path      Path to render
+ * @param  {Object} variables Object mapping variable to actual values
```

Comment:
same "folders" ?

Reply:
No, here it is not the magic folders but folders definitions coming from the konnector manifest and the associated values, coming from the context.

Reply:
Done